rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,143 @@
+## Summary
+[summary]: #summary
+
+This rfc proposes a new tool dubbed "minaform" for reliably creating and managing QA nets and production deployments. Our existing setup is very hacky and requires knowledge of many tools to use effectively. The intention with minaform is to build a flexible enough tool that resolves major pain points with the existing setup, and standardizes a single tool for folks to learn/understand.
+
+## Motivation
+[motivation]: #motivation
+
+The goal is to standardize community-oriented and internal qa networks with a simple tool to ensure that the process for creating and deploying a new testnet is code-reviewed, documented, and standard across all deployment scenarios. As of this RFC, additional requirements for community members themselves or external partners to make use of this tooling is out-of-scope but the goal is that this implementation will be easier to maintain and extend as needs arise.
+
+The following issues with the existing system were reported by at least one person as "pain points" that consume time/debugging effort when deploying networks:
+
+1) Production Networks vs. QA Nets
+    - The process for deploying production testnets vs. qa nets has diverged in a few places and this means our day-to-day testing does not cover all of the services we need to deploy for public networks.
+2) Determining network health is difficult/unclear/inconsistent
```

Comment:
Cool, I like the idea of providing links to grafana dashboards and other useful bits of post-deploy guidance (thinking as Terraform outputs?) following `apply`.

May also be worth allocating dedicated prometheus and grafana instances for each testnet (w/ templated and generated grafana dashboards + publicly accessible dns names) to help avoid spamming the grafana labs board or get rid of it all together.

Also seeing if there's anything we can make use of in terms of Helm *Notes* when it comes to post-deployment steps/guidance could be worthwhile.